tests/fetchStats.test.js

```diff
@@ -92,10 +96,23 @@ const error = {
 
 const mock = new MockAdapter(axios);
 
+/**
+ * Mocks the GraphQL API based on certain conditions.
+ *
+ * @param {*} config Axios config object.
+ * @returns Axios response object.
+ */
+const mockData = (config) => {
+  // If year is 2003, return dataYear.
+  if (config.data.includes("contributionsCollection(from: $starttime)"))
+    return [200, dataYear];
+  return [200, data];
+};
+
 beforeEach(() => {
   mock
     .onPost("https://api.github.com/graphql")
-    .replyOnce(200, data)
+    .replyOnce(mockData)
```

Comment:
Here I now handle the year data return conditional switching using the mockData function.